### PR TITLE
feat(vesting): withdraw from active schedule

### PR DESCRIPTION
**Pull request by epanchee**

### CODE REVIEW

File: contracts/tokenomics/vesting/src/contract.rs [hunk 1]
```diff
@@ -320,26 +326,134 @@ fn compute_available_amount(current_time: u64, vesting_info: &VestingInfo) -> St
...
+                end_point: Some(VestingSchedulePoint {
+                    time: end_point.time,
+                    amount: end_point.amount - sch_unlocked_amount - amount,
+                }),
+            }
+        } else {
+            VestingSchedule {
+                start_point: VestingSchedulePoint {
+                    time: block_time + 1,
+                    amount: schedule.start_point.amount - sch_unlocked_amount - amount,
```
Comment (ShadoySV) [inline contracts/tokenomics/vesting/src/contract.rs:426]: User could withdraw starting amount at this point. Let's exclude one-time schedules

**Review by ShadoySV — CHANGES_REQUESTED** [review]

File: contracts/tokenomics/vesting/src/contract.rs [hunk 2]
```diff
@@ -320,26 +326,134 @@ fn compute_available_amount(current_time: u64, vesting_info: &VestingInfo) -> St
...
+            .unwrap_or(schedule.start_point.amount)
+            .checked_sub(sch_unlocked_amount)?;
+        if amount >= amount_left {
+            return Err(ContractError::NotEnoughTokens(amount_left));
+        }
+
+        new_schedule = if let Some(end_point) = &schedule.end_point {
+            VestingSchedule {
+                start_point: VestingSchedulePoint {
+                    time: block_time + 1,
```
Comment (ShadoySV) [inline contracts/tokenomics/vesting/src/contract.rs:414]: Adding one makes a gap between result schedules

File: packages/astroport/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "astroport"
-version = "2.1.0"
+version = "2.1.1"
```
Comment (ShadoySV) [inline packages/astroport/Cargo.toml:3]:
```suggestion
version = "2.2.0"
```
because functionality was added

**Review by ShadoySV — CHANGES_REQUESTED** [review]

File: contracts/tokenomics/vesting/src/contract.rs (hunk 1 above)
Comment (epanchee) [reply contracts/tokenomics/vesting/src/contract.rs]: True. Fixed

File: contracts/tokenomics/vesting/src/contract.rs (hunk 2 above)
Comment (epanchee) [reply contracts/tokenomics/vesting/src/contract.rs]: Ok, removed this gap

File: contracts/tokenomics/vesting/src/contract.rs [hunk 4]
```diff
@@ -320,26 +326,126 @@ fn compute_available_amount(current_time: u64, vesting_info: &VestingInfo) -> St
...
+    let acc = deps.api.addr_validate(&account)?;
+    let mut vesting_info = VESTING_INFO.load(deps.storage, &acc)?;
+    let block_time = env.block.time.seconds();
+
+    let mut active_schedules = vesting_info.schedules.iter_mut().filter(|schedule| {
+        if let Some(end_point) = schedule.end_point {
+            block_time >= schedule.start_point.time && block_time < end_point.time
+        } else {
+            block_time >= schedule.start_point.time
+        }
```
Comment (ShadoySV) [inline contracts/tokenomics/vesting/src/contract.rs:390]: We should be able to withdraw even if there are any one-time schedules. Also let's allow withdrawing when `block_time < schedule.start_point.time`

**Review by ShadoySV — CHANGES_REQUESTED** [review]

**Review by ShadoySV — APPROVED** [review]

**Review by P-Yevhenii — APPROVED** [review]

**Pull request merged**